apps/web/src/api/hooks/use-filter-templates.ts

```diff
@@ -0,0 +1,49 @@
+import { useQuery } from '@tanstack/react-query';
+import { INotificationTemplate } from '@novu/shared';
+
+import { getNotificationsList } from '../notifications';
+import { useEnvController } from '../../store/use-env-controller';
+
+export function useFilterTemplates(searchQuery = '', page = 0) {
+  const { environment } = useEnvController();
+  const { data: tempData, isLoading: isTempLoading } = useQuery<{
+    data: INotificationTemplate[];
+    totalCount: number;
+    pageSize: number;
+  }>(['notificationsList', environment?._id, page], () => getNotificationsList(page), {
+    keepPreviousData: true,
+  });
+
+  const { data, isLoading } = useQuery<{
+    data: INotificationTemplate[];
+    totalCount: number;
+    pageSize: number;
+  }>(
+    ['notificationsList', environment?._id, 0, isTempLoading ? 10 : tempData?.totalCount],
+    () => getNotificationsList(0, isTempLoading ? 10 : tempData?.totalCount),
+    {
+      keepPreviousData: true,
+    }
+  );
```

Comment:
I couldn't understand the flow.. correct me if I'm wrong...
1. I fetch all the templates normally
2. When the search query is not empty, I should use the QueryClient API to get notifications from the cache?

Reply:
yes, that's correct :) like when you fetch the notifications for page n, then n+1, all they should be stored (for x amount of time) in the query cache, meaning you can retrieve it and do the filtering

Reply:
> yes, that's correct :) like when you fetch the notifications for page n, then n+1, all they should be stored (for x amount of time) in the query cache, meaning you can retrieve it and do the filtering

so that would apply search filtering to only the navigated template pages? wouldn't that make the search not trully valid

Reply:
I'm sorry for taking a lot of time on this... but also wanted to ask how to write e2e tests... for this?